### DISCUSSION TITLE
1 Project with 3 developpers

### DISCUSSION THREAD

**Discussion by Marcoparigi** [symfony/symfony#45805] [Q&A] [answered]
Hi all,
I have a live project made with 2.7 symfony version. It is live since 2016.
I was alone working on it.
But now we have hired 2 more people to maintain and enhance the  application.
We need to work as a dev team using versioning.
We are searching the best way to implement git with vscode with the following information :
_ 1 Linux server hosting the framework,the web server and the database.
_ Each developper is using its own windows virtual machine with notepad++ and sftp.

As you can see working like this is not easy because of no versionning and high risk to overwrite other developper team member.

We want to use vscode with git (git installed on the existing server)

Question : How each dev team member can test his own commited code before pushing it to the server ?
Do we need to install on each dev team member virtual machine wamp to use a local web server ? (Symfony 2.7 doesn't include the local web server 😢)

Thank you for reading and long life to Symfony !! 😉

**Comment by GromNaN** [comment, marked as answer]
Hi, using Git is of course a good idea; and a necessity.

For code hosting and collaboration, I would recommend using a centralized server like GitHub or Gitlab. Private projects can be hosted for free.

If you all want to work on the same remote server, you can create a Linux user and an vhost for each developer. So that everyone as its own workspace.

**Reply by Marcoparigi** [reply]
Thank you GromNan for your interest and your answer.
Our main problem is what is the best way to test locally our cloned repository when applying changes before pushing.
I am still searching throw youtube videos and some guys talked about Live server extension with PHP server extension for vscode.
I didn't had time to test this but maybe it can solve my problem.
Do you agree with this kind of extension ?
Thanks a lot
